UC NetCode 2024 Fall/Assets/App/Resource/Scripts/Player/PlayerInfoData.cs

```diff
@@ -0,0 +1,66 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Unity.Collections;
+using Unity.Netcode;
+using System;
+
+public struct PlayerInfoData : INetworkSerializable, IEquatable<PlayerInfoData>
+{
+    public ulong _clientId;
+    public FixedString64Bytes _name;
+    public bool _isPlayerReady;
+    public Color _colorId;
+
+    public PlayerInfoData(ulong id)
+    {
+        _clientId = id;
+        _name = "";
+        _isPlayerReady = false;
+        _colorId = Color.cyan;
+    }
+
+    public void NetworkSerialize<T>(BufferSerializer<T> serializer) where T : IReaderWriter
+    {
+        if (serializer.IsReader)
+        {
+            var reader = serializer.GetFastBufferReader();
+            reader.ReadValueSafe(out _clientId);
+            reader.ReadValueSafe(out _name);
+            reader.ReadValueSafe(out _isPlayerReady);
+            reader.ReadValueSafe(out _colorId);
+        }
+        else
+        {
+            var writer = serializer.GetFastBufferWriter();
+            writer.WriteValueSafe(_clientId);
+            writer.WriteValueSafe(_name);
+            writer.WriteValueSafe(_isPlayerReady);
+            writer.WriteValueSafe(_colorId);
+        }
+    }
+
+    public bool Equals(PlayerInfoData other)
+    {
+        return _clientId == other._clientId && _name.Equals(other._name) && other._isPlayerReady && _colorId.Equals(other._colorId);
```

Comment:
I spent most of my evening  trying to hunt this guy down and should really help you feel a little more relief that it also gave me some trouble haha. I didn't want to let you down so I made sure to figure this out for you. This was a head scratcher and the biggest clue that lead me down this bug was seeing what happened at the moment of when it was removing player data, 

We did get a successful disconnect from the  player so it was very confusing when at first glace everything is where it should be and calling the correct functions.

 So I figured it had to be a deeper root issue and I noticed in our custom structure we have some Equals set. Things like different name or player colors sometimes  get out of sync, So we trim this down to look at only just the client ID, where that specific information is what really should be reliable. 
 
 
```suggestion
        return _clientId == other._clientId;
```

Reply:
Hi, I'm so sorry for looking at this so late, is this the comment with the issues or is there more under "reviewed changes" because I can't see anything in there. I've spent most of this semester confused by my work as I still haven't even been able to do much with my commits because of conflicts, is there anyway you could help with those as well?